asyncscope.md

```diff
@@ -280,164 +332,314 @@ loop. We use `counting_scope` to limit the lifetime of the request handling with
-  counting_scope();
-  ~counting_scope();
-  counting_scope(const self_t&) = delete;
-  counting_scope(self_t&&) = delete;
-  self_t& operator=(const self_t&) = delete;
-  self_t& operator=(self_t&&) = delete;
-
-  void
-  /*@@_implementation-defined_@@*/ /*@@_customization-point_@@*/(
-    decays_to<self_t> auto&&, spawn_t, sender_to<@@_spawn-receiver_@@> auto&& s) noexcept;
-
-  template <sender_to<@@_spawn-future-receiver_@@> S>
-  @@_spawn-future-sender_@@<S>
-  /*@@_implementation-defined_@@*/ /*@@_customization-point_@@*/(
-    decays_to<self_t> auto&&, spawn_future_t, S&& s) noexcept;
-
-  template <sender S>
-  @@_nest-sender_@@<S>
-  /*@@_implementation-defined_@@*/ /*@@_customization-point_@@*/(
-    decays_to<self_t> auto&&, nest_t, S&& s) noexcept;
+    counting_scope() noexcept;
+    ~counting_scope();
+
+    counting_scope(const counting_scope&) = delete;
+    counting_scope(counting_scope&&) = delete;
+    counting_scope& operator=(const counting_scope&) = delete;
+    counting_scope& operator=(counting_scope&&) = delete;
+
+    template <sender S>
+    struct @@_nest-sender_@@; // @@_exposition-only_@@
+
+    template <sender S>
+    [[nodiscard]] @@_nest-sender_@@<std::remove_cvref_t<S>> nest(S&& s) & noexcept(
+            std::is_nothrow_constructible_v<std::remove_cvref_t<S>, S>);
+
+    struct @@_join-sender_@@; // @@_exposition-only_@@
+
+    [[nodiscard]] @@_join-sender_@@ join() noexcept;
+
+    // observers in the spirit of std::weak_ptr<T>::expired() and std::shared_ptr<T>::use_count();
+    // the values must be correct when computed but may be stale by the time they can be observed
+
+    [[nodiscard]] bool joined() const noexcept;
+
+    [[nodiscard]] bool join_started() const noexcept;
+
+    [[nodiscard]] size_t use_count() const noexcept;
 };
 ```
 
-## Lifetime
+## `execution::nest`
 
-The `counting_scope` keeps a counter of how many spawned _`async-function`_ s have not completed.
+```c++
+template <sender Sender>
+auto nest(Sender&& snd, auto&& scope) noexcept(noexcept(scope.nest(std::forward<Sender>(snd)))
+    -> decltype(scope.nest(std::forward<Sender>(snd)));
+```
 
-## `spawn()`
+Attempts to associate the given sender with the given scope in a scope-defined way. When successful, the return value
+is an "associated sender" with the same behaviour and possible completions as the input sender, plus the additional,
+scope-specific behaviours that are necessary to implement the scope's bookkeeping policy. When the attempt fails,
+`nest()` may either eagerly throw an exception, or return a "unassociated sender" that, when started, unconditionally
+completes with `set_stopped()`.
+
+A call to `nest()` does not start the given sender and is not expected to incur allocations.
+
+When `nest()` returns an associated sender:
+
+ - connecting and starting the associated sender connects and starts the given sender;
+ - the associated sender is multi-shot if the input sender is multi-shot and single-shot otherwise; and
```

Comment:
do we explain what "multi-shot" means?

Reply:
Yeah, it's defined in P2300: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2300r7.html#design-shot